scvi/train/_trainingplans.py

```diff
@@ -593,20 +593,31 @@ class PyroTrainingPlan(pl.LightningModule):
     optim
         A Pyro optimizer, e.g., :class:`~pyro.optim.Adam`. If `None`,
         defaults to Adam optimizer with a learning rate of `1e-3`.
+    optim_kwargs
+        Keyword arguments for default optimiser (pyro.optim.ClippedAdam).
     """
 
     def __init__(
         self,
         pyro_module: PyroBaseModuleClass,
         loss_fn: Optional[pyro.infer.ELBO] = None,
         optim: Optional[pyro.optim.PyroOptim] = None,
+        optim_kwargs: Optional[dict] = None,
     ):
         super().__init__()
         self.module = pyro_module
         self._n_obs_training = None
 
+        optim_kwargs = optim_kwargs if isinstance(optim_kwargs, dict) else dict()
+        if "lr" not in optim_kwargs.keys():
+            optim_kwargs.update({"lr": 1e-3})
+        if "clip_norm" not in optim_kwargs.keys():
+            optim_kwargs.update({"clip_norm": 200})
+
```

Comment:
what if user passes in an optim that doesn't take clip norm? shouldn't this check first if lr and clip norm are in the signature of the optim init?

Reply:
optim_kwargs are only used for default optimiser.